### ISSUE TITLE
Sprint - Nov 25 - December 6

### ISSUE THREAD

**Issue by Phanatic** [PostHog/posthog#26305]
# Global Sprint Planning

## 3 things that might take us down

1. EOY on call schedule 
2. Django thrashing (@benjackwhite assemble djangovengers)
3.

## Urgent incident follow-ups

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
# Team ___

**Support hero:** ___

## Retro

<!-- Grab the high and low priority items from last time and add whether that item was completed or not -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

-

### Low priority / side quests

-

```

**Comment by Phanatic** [comment]
# Team Surveys

## Retro

✅ Adaptive sampling for collecting survey responses.
✅ Surveys bug fixes
✅ Feature preview release of no-code experiments
✅ Documentation for no-code experiments.

---

## OKRs

1. Implement Enterprise-ready features for Surveys
2. Increase adoption of Surveys.

### High priority

#### Features:
1. https://github.com/PostHog/posthog/issues/25432
2. https://github.com/PostHog/posthog/issues/25477

#### Investigation
* Surveys via emails.

**Comment by jurajmajerik** [comment]
# Team Experiments 🧪 

## Retro
- 🟢 @danielbachhuber joined the team! 🥳 
	- @raquelmsmith is onboarding him in person
	- First PRs in: UI improvements to Experiment goal, onboarding flow, up to 10 secondary metrics
	- Running first experiment - select multiple products in the product screen 

- 🟢 [priority_customer]
	- Holdout groups implemented @neilkakkar @jurajmajerik 
	- Added support for 10 secondary metrics @danielbachhuber 
	- Backend for reusable metrics - 90% done @neilkakkar 
	- We have issue where we keep track of any new requests: https://github.com/PostHog/product-internal/issues/667 
	- The plan is to over communicate with them in their Slack channel to keep them engaged/gather feedback/show we're proactive.

- 🟡 Continued on HogQL for Experiments @jurajmajerik  (https://github.com/PostHog/posthog/issues/24914)
	- Opted for a large refactor of the Experiments logic
	- Shared components (selectors) with Product Analytics, but tightly coupled with their logic. Convoluted state flows, hard to reason about.
	- Forked these components - we want them to work with our logic
	- We want to be in control of these components - make sure the statistics always work correctly for any goal metrics.
	- This is taking extra time/delaying the HogQL completion, but worthwhile, will enable us to move faster.

- 🟢 Review statistics with a contractor @jurajmajerik 
	- Latest feedback incorporated, seem to be done for now - all main metric types supported.
	- Next step: plug these methods into our APIs and write tests
	- Might need some tweak later

- 🟢 Support hero
	- Experiments queue is set up now
	- Workload manageable

## Sprint goals
- Complete the onboarding list (mostly UI improvements) @danielbachhuber 
- Data Warehouse integration for [priority-customer] - pending information from them - start collecting requirements @danielbachhuber 
- HogQL rewrite @jurajmajerik 
	- Finish Experiments logic refactor
	- Migrate Experiment records
	- Test behind feature flag
- Support hero @jurajmajerik

**Comment by mariusandra** [comment]
# Team Cut Delivery Prices

- Support: Marius

## Retro

- CDP focus on ad conversion platforms integrations
  - @mariusandra client side destinations will be ready by end of week
  - @benjackwhite 
  - @oliverb123 error tracking
  - @MarconLP built functions for most server side ads destinations


## OKRs
* Widespread usage of the new CDP 
* Messaging V1 
* Ingestion is rock solid

## Sprint focus

### High priority

-

**Comment by EDsCODE** [comment]
# Team Data Storehouse
### OKR Q4 2024

### Objective

**Query 3000**

* **Key Results:**
    * Autocomplete
	* Increase general BI experience/product https://github.com/PostHog/meta/issues/157
	* Declutter the data warehouse UI and make the features intuitive to find

**Data Modeling MVP**

* **Key Results:**
	* Infrastructure decided and implemented
	* Integrating external data with feature flags
	* External data everywhere in insights/persons/cohorts
	* Get billing team to use modeling in posthog for their invoices_with_annual table
 
### Retro
- [ ] Release dashboard insight variables @Gilbert09 
- [ ] BigQuery source release to everyone @Gilbert09 
- [ ] Reduce temporal-data-warehouse pod size (memory) @Gilbert09 
	- Became main focus + starting a rebuild of import pipeline that’s much simpler
- [ ] SQL editor 3000 MVP @EDsCODE 
	- most of this will be good for internal use by friday
- [ ] SPMC Batch Exports @tomasfarias
	- backed up by incident
- [ ] Splitting persons batch exports @tomasfarias
	- Not going to do. Initially wanted to split these queries but Tim’s query performance improvement of persons exports works well
- [x] HogQL ambiguous query error when joining tables with the same field names @rossgray
- [x] dataviz tests @rossgray
- [x] view validation @rossgray
- [ ] One of: Backfill UI, test connection button, or sessions/groups batch exports @rossgray


### Priority

- [ ] Data import rebuild @Gilbert09 
- [ ] SQL editor 3000 finish MVP section + move to new priorities @EDsCODE 
- [ ] Single producer multiple consumer Batch Exports @tomasfarias
- [ ] Query cancellation @rossgray
- [ ] Autoscaling based on metrics @rossgray @tomasfarias

**Comment by dmarticus** [comment]
# Team Flags

## Support

Starting support hero rotation, Dylan will go 11-18 to 11-25, Haven will take 11-25 to 12-2 (dylan supporting as needed).

## Retro

### Things we got done

- [x] Breakage from lack of type safety in List logic
    - shipped customer-facing issue that our types didn't really help us catch
    - fixed issue, fixed types around this domain so that we don't get bitten by this thing going forward
- [x] Made changes to metrics (tracking `creation_context`) and Sentry dashboard + issue assigning
- [x] [RFC proposed](https://github.com/PostHog/product-internal/pull/677) on `Feature Management` changes sidebar UI

---

## Sprint

### High priority

- [ ] testing/productionizing new `/flags` service @dmarticus 
    - [ ] functional testing via django-compat integration tests
    - [ ] "real" testing via mirroring traffic to new service (meeting with ingestion team 11-20 to go through this)
    - [ ] generally ironing out service (centralizing error handling, reducing memory usage overhead with smarter copying/cloning)
- [ ] Start work on feature management UI @havenbarnes 
- [ ] land v0 of flag staleness feature @havenbarnes 
- [ ] Track `feature flag called` events in usage reports so we can track up who set up flags successfully @dmarticus 

#### Side Quests

- [ ] Continue to make sentry more actionable and useful @havenbarnes 
- [ ] Figure out how to cut off the flags bit of `/decide` (and, in future `/flags`) when folks go over – right now we don't stop serving flags when overage happens @dmarticus 
- [ ] Continue to look at growth metrics and see what other properties we can add to better understand our growth @dmarticus 

#### Things to chew on

- [ ] Can we know when people are starting to make new features?

## OKRs

1. Ship new feature flags in production and accomplish the 10x scale + 2x speed boost performance targets (ideally decreases infra spend by N% as well, will try to figure out a more specific target going forward)
2. Maintain current /decide service and make optimizations.

**Comment by fuziontech** [comment]
# Team ClickHouse

## OKR Q2 2024

### Objective

**Goals for Q4:**

- Improve elasticity and flexibility of our data store by putting all our data in Iceberg
    - [Work with Altinity](https://docs.google.com/document/d/12icx71-_kV0aICP3igtOc0di_yrey_W82HzbadtNPBY/edit?usp=sharing) effectively to ship read path for Iceberg on ClickHouse - Brett Hoerner
    - Setup infrastructure to ship all of our data to Iceberg on S3 - James Greenhill
    - Shipping Query logs to Iceberg - James Greenhill
- Continue improving CH operational expertise
    - Upgrade to a later version of Clickhouse - James Greenhill
    - Capacity planning - James Greenhill
    - Automation - Daniel Escribano
    - Put some of the basic mitigation operations in runbooks - Daniel Escribano
- Schema management
    - Tool for schema migration (coordinator schemas) - Daniel Escribano
    - Tool for long running mutations - Daniel Escribano
- Continued investment in performance
    - Tooling for other teams to understand which queries are slow and why - Ted Kaemming
    - Investigate variability of queries - Ted Kaemming
    - Per-team limits on queries/query complexity (needs product work) - Ted Kaemming

## Board
https://github.com/orgs/PostHog/projects/85/views/2

## Retro

### 🔥 Incidents 🔥 
- [#inc-2024-11-14-clickhouse-eu-replication-issues](https://posthog.slack.com/archives/C080VJV4D0S)
- [#inc-2024-11-11-batch-exports-backlog-in-us](https://posthog.slack.com/archives/C080D7APEMS)

### Last Sprint's Goals
- Continue 24.8 upgrade and test fix @fuziontech ✅ 
- Ongoing Delete tasks (support) @fuziontech ✅ 
- Remove unused Materialized Columns @tkaemming ⌛ 
	- Dropping indexes is timing out, need to figure out why that's happening 🤔 
- Reshard US into 8 shards 3 replicas w/ coordinators @Daesgar  ⏳  
	- Already in 2024 for partitions - might be complete this week! Definitely next week.
	- Only 1 replica currently
- Snapshot querying of Iceberg tables proof of concept @bretthoerner @fuziontech ❌ 
	- Working on `_timestamp` table for now which meets the needs of BE today
- Fix maintenance tasks such as snapshot pruning @bretthoerner  ✅ 
	- Was a bunch of spark tunables that are in TF and have been working for a week
- [Document pyspark/EMR runbook stuff in general before](https://github.com/PostHog/runbooks/pull/102) 🛫  @bretthoerner ✅ 

## This Sprint's Goals 

### 🦃 day plans
@Daesgar Will cover Wed, 🦃-day, Friday
@tkaemming & @fuziontech will be in the desert frying a 🦃 
@bretthoerner - Enjoying Funemployment (Last day is Wednesday)

### TL;DR

- Validate cluster data parity between old and new clusters @fuziontech 
- Upgrade clusters to 24.8 (US, EU, new-US, Dev) @fuziontech @Daesgar 
- Anticipating new 24.8 cluster w/ real lightweight deletes - fix deletes @fuziontech 
- Migrate to new 8 shard lower cost cluster w/ coordinator nodes @Daesgar 
- Investigate how to speed up index dropping for Mat Cols @tkaemming 
- (low priority) Expose Iceberg table for testing @fuziontech 
	- Current version this will be quite slow
	- New 24.8 version will be quite fast
- Better caching of object store bits for Iceberg (Working in CH codebase) @bretthoerner 
	- Code does a HEAD request for every file even if it's in cache for files that don't change

**Comment by aspicer** [comment]
# Team Product Analytics
🎉 Peter Joining Dec 9th!
**Support hero:** @thmsobrmlr @anirudhpillai 

## Retro

Anirudh (Support Nov 11 - 15)

- 🟢 Alerts currently rolled out to 20% of users, sprint goal to roll out to 100%
    - Stability Fixes
- 🟢 RFC for Saved Filters / Saved HogQL Snippets
     - PR up for experimenting with different approaches 
 - 🟢 (Unplanned) Added trends with breakdowns to alerts
- (Reprioritized) Add slack messaging alerts
- (Side Quest - Repriotized) Prometheus push gateway for gathering metrics from Celery

Thomas
- 🟢 Filter Removal (Dashboard templates)
- 🟢 Custom colors on funnels / trends
- 🟢 Chart reloading when disable breakdowns

Sandy (Support Nov 18 - 22)

-🟢 Password protection (for third party shares) https://github.com/PostHog/posthog/issues/10869
- (On Hold - Thomas investigating if it overlaps with his work on event metadata) Investigate getting date first seen onto a person https://github.com/PostHog/posthog/issues/11952 https://github.com/PostHog/posthog/issues/3031 

## Upcoming Sprint
Anirudh (Support second week)
- Alert Comms / Launch
- Support for launch / fast-follow features for alerts
- Cached dashboards (https://github.com/PostHog/posthog/issues/26309)

Thomas (Support First Week)
 - Migrate api endpoints to use hogql queries
 - Look into filters removal broadly
 - Fix lifecycle charts with personless events

Sandy
 - Finish Cohort Migration
 - Spec out funnel improvements for December https://github.com/PostHog/posthog/issues/17313
 
Anna
 - Group Analytics Research, generate plan for Q1

**Comment by zlwaterfield** [comment]
# Team Growth

## Retro
<details>  
<summary>Retro items</summary>

- @zlwaterfield 
	 - RBAC
	 	- [x] Get access models shipped
	 	- [ ] Get base middleware implements
	 	- [ ] Pick first product and get the base RBAC setup for it
	- [x] Get Teams Trials fully live
	- [x] Ship lowering of self serve credits requirements
	- [x] Finish and ship project limits API checks
	- [x] Onboarding for Surbhi
	- if time: 
		- [ ] (from last time) Make a few improvements to the technical exercise (based on using it for 2 weeks) 
		- [x] (from last time) Get the in-app credit card entry experiment shipped 
		- [x] Review existing billing metrics, hookup to billing on call, see if we should add others
		- [x] customer io transaction api
		- [ ] More billing metrics for on call
- @patricio-posthog 
	- [x] Continue with forecasting bugs, trying to understand what's happening and how to solve it.
	- [x] (From last time): Turn off daily invoice sync to rely 100% on webhooks.
	- [ ] Try to optimize `/billing` endpoint, maybe separating forecasting into its own endpoint (after fixing it and understand how it works)
	Extra things if there is time:
	- [x] Start to analyze forecasting and see its accuracy after each period.
- @raquelmsmith 
  - [ ] Get product analytics dashboard templates out again
  - [ ] follow up on replay templates / reengagement campaign
  - [x] Daniel in-person onboarding
  - [x] Multi-product-select in onboarding (with Daniel)
- @surbhi-posthog 
  - [x] onboarding
  - [x] first 3 prs - sign up page bugs 
</details>  

## Q4 Goals
✅=finished 🟡=in progress 🔴=won't finish ⚪=not started
1. 🟡 Get to a 4/10 in Enterprise platform support (Zach)
	- 🟡 Release Enterprise addon and iterate with sales to find a pricing model that makes sense
	- 🟡 Build RBAC
2. 🟡 Make revenue & forecasting more reliable (Patricio)
	- 🟡 Rejig how we do sync_invoices so we don’t see big drops - respond to webhooks instead of large syncs
	- ⚪ Evaluate temporal vs celery for usage reports and billing tasks so we can adhere to rate limits better
	- ⚪ Side hustle, if time permits: Rejig how anonymous / identified events are billed, as the current implementation is confusing.
3. 🟡 Activation: reverse trials, analytics onboarding flow, understand CC activation (Raquel)
	- ⚪ Dig into the data, watch replays to figure out why CC activation rate has been falling
	- ⚪ Add reverse trial to onboarding
	- ✅ Add reverse trial to billing page for teams

## This sprint

Time off: @raquelmsmith, @zlwaterfield and @surbhi-posthog in Palm Springs for offsite.

- @zlwaterfield 
	 - RBAC
		- [ ] Get middleware PR tested and shipped
	 	- [ ] Get RBAC UI shipped for 1 product 
	- [ ] Improve how we show value to teams addon customers when unsub'd or cancelling trial (think about how to do this for all products)
	- [ ] Talk to DW about materialzed views
	- [ ] Talk to Mine/Joe about better startup plan labels
	- [ ] Continue talking to Stripe PM about billing problems we are running into
- @surbhi-posthog 
	-  [ ] Review [exiting ICP research](https://posthog.slack.com/archives/C043VJ93L3B/p1723471287020649)
	- [ ] Connect with Mine about ICP data being sync'd into PostHog
	- [ ] Do some analysis on our metrics through signup, onboarding, converting to paid based on our ICP scoring data
	- [ ] Suggest (and maybe implement) changes to the onboarding product experience based on learnings from the analysis.
- @patricio-posthog 
	- [ ] Forecasting: 1 - Pato: 0. Still fighting this battle.
	- [ ] New pricing for queries (API) and mobile

**Comment by robbie-c** [comment]
# Team web analytics

**Support hero:** @robbie-c  + SDK support hero

## Retro

Mike --> Rafael

Created SDK Support Hero

#### Tasks
🔴 Finish cookieless prototype
🔴 Make a start on page reports / core web vitals
🟢 Custom channel types in queries
🟢 Fancy UI for custom channel types

## OKR

* Continue to improve performance
* Build a small amount of more advanced marketing features, to measure interest
* Cookieless tracking
* Improve the NextJS experience
* More page-level insights
* Improve the documentation, and link to it more from the app

### High priority
Robbie
- Onboard Rafael
- Support hero
- Finish cookieless prototype

### Stretch goals
Mike
- https://github.com/PostHog/posthog/issues/26003

**Comment by Twixes** [comment]
# Team AI PM

## Retro

> ### High priority
> 
> - 🟢 Results summarization
> - 🟢 Generalize Session Replay's OpenAI data processing opt-in (to be determined with @fraserhopper) @Twixes
> - 🟢 Stream feedback on the agent's current task in conversation @Twixes – initial experience is released
> - ~~PostHog dashboard for AI product manager use @Twixes~~ – deprioritized in favor of quality
> - 🟢 Tune Trends generation for more relevant results (smarter application of breakdowns, better property value examples) @skoob13 – work in progress, but good progress
> - 🟡 Handle errors (agent recursion limit + graceful handling of e.g. ClickHouse errors) @skoob13
> 
> ### Low priority / side quests
> 
> - 🟢 Results evaluation framework - a benchmark @skoob13 – prioritized due to quality focus
> - ~~Environments out as a feature preview @Twixes~~ – deprioritized in favor of AI PM quality

## OKR

1. OKR, status (red/yellow/green) and action points if yellow/red


### High priority

- Fix response streaming in production @skoob13 
- Make agent self-aware when it's not capable enough to answer the question _yet_ @Twixes 
- Max AI rolled out as feature preview
- SQL generation MVP
- All the insights types
- Memory, prewarmed
### Low priority / side quests

- Environments out as a feature preview @Twixes
